app/workers/pull_request_monitor_handlers/path_based_labeler.rb

```diff
@@ -0,0 +1,27 @@
+class PullRequestMonitorHandlers::PathBasedLabeler
+  include Sidekiq::Worker
+  sidekiq_options :queue => :miq_bot
+
+  include BranchWorkerMixin
+
+  def perform(branch_id)
+    return unless find_branch(branch_id, :pr)
+    return unless verify_branch_enabled
+
+    process_branch
+  end
+
+  private
+
+  def process_branch
+    label_rules.each do |rule|
+      if diff_file_names.any? { |file_name| file_name =~ Regexp.new(rule.pattern) }
+        GithubService.add_labels_to_an_issue(fq_repo_name, pr_number, [rule.label])
```

Comment:
> Thinking about this...since GithubService::Issue is a reusable class now, perhaps we just fetch that issue, then call add_labels directly on that?

That entirely depends on your previous comment; I didn't think so at first given that you'd then be doing a fetch before doing something that I *assumed* was idempotent (but, as you said, I should probably verify first 😅)